Added convert command

shashigharti opened the pull request:
Added convert command
Added tests

- fixes #1170

---

Please make sure that all the checks pass. Please add here any additional information regarding this pull request. It's highly recommended that you link this PR to an issue (please create one if it doesn't exist for this PR)

shashigharti commented:
@roll it is ready for review!

roll commented on frictionless/program/convert.py:
```diff
@@ -0,0 +1,53 @@
+        typer.secho(message, err=True, fg=typer.colors.RED, bold=True)
+        raise typer.Exit(1)
+
+    # Initialize Package
```
Sorry @shashigharti,

the initial issue description was quite confusing as a I see now. What I meant that we can have metadata conversion feature actually for all types of metadata (package/resource/schema/etc).

So if we do something like this:

```python
 try:
        metadata_type = Detector.detect_descriptor(source)
        if metadata_type == 'package':
          metadata = Package.from_descriptor(source)
        elif metadata_type == 'resource':
          metadata = Resource.from_description(source)
        # and more types
    except Exception as exception:
        typer.secho(str(exception), err=True, fg=typer.colors.RED, bold=True)
        raise typer.Exit(1)
```

Then we can do:

```python
if yaml:
        content = metadata.to_yaml(path)
        typer.secho(content)
        raise typer.Exit()
```

shashigharti replied:
It only detects resource and package only. 
metadata_type = Detector.detect_descriptor(source)

what do you suggest?

roll commented on frictionless/program/convert.py:
```diff
@@ -0,0 +1,53 @@
+        raise typer.Exit(1)
+
+    # Return yaml
+    if yaml:
```
Let's also support `json`. For example, use case:

```
frictionless convert schema.yaml --json
```

roll commented on frictionless/program/convert.py:
```diff
@@ -0,0 +1,53 @@
+
+    # Validate input
+    if not source:
+        message = 'Providing package "source" is required'
```
Providing "source" is required (as it can be other types not only package)

roll commented on frictionless/program/convert.py:
```diff
@@ -0,0 +1,53 @@
+        raise typer.Exit()
+
+    # Return ER Diagram
+    if er_diagram:
```
This one will work only for `metadata_type == 'package'`

roll reviewed (changes requested):
It's a good start and we can extend it to support all the metadata types

shashigharti committed:
Changed convert command to include all metadata classes

Included additional metadata classes
Added markdown template for metadata classes whose markdown template were missing
Added tests

shashigharti commented:
@roll Thank you! I have made the changes and it is ready for the review.

roll reviewed (approved):
Great thanks!

roll commented:
Thanks for adding those convert tests for different metadata types -- it will be really useful to catch regressions :+1:
